### PR TITLE
Wrap token amounts with BigNumber, generate spendTx by SDK

### PR DESCRIPTION
**Pull request by davidyuk**

### CODE REVIEW

File: src/lib/constants.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// eslint-disable-next-line import/prefer-default-export
+export const MAGNITUDE = 4;
```
Comment (davidyuk) [inline src/lib/constants.js]: We should change it to 18 when testnet will have enough tokens.

File: src/main.js [hunk 2]
```diff
@@ -33,7 +35,8 @@ Vue.use(VeeValidate, {
         min_value_exclusive: (field, [min]) => `This field must be more than ${min}`,
         max_value: (field, [max]) => `This field must be ${max} or less`,
         not_in: () => 'This field must be a valid value',
-        decimal: () => 'This field must be numeric and may contain decimal points',
+        decimal: (field, [decimals = '*'] = []) =>
+          `This field must be numeric and may contain ${!decimals || decimals === '*' ? '' : decimals} decimal points`,
```
Comment (davidyuk) [inline src/main.js]: Taken form here: https://github.com/baianat/vee-validate/blob/1d99883c131cf4bfe9b2e1155e2260d3396e1ca1/locale/en.js#L16

File: src/lib/constants.js (hunk 1 above)
Comment (edwarddikgale) [reply src/lib/constants.js]: Hey @davidyuk  in ApproveTransaction.vue what happens if transaction is null? I don't see you checking for this. Is this a possibilit? is vue forgiving for scenarios like this? Other code looks splendid to me

Comment (edwarddikgale) [reply src/lib/constants.js]: Great, looks good

**Pull request merged**